scripts/scr_cheatcode/scr_cheatcode.gml

```diff
@@ -8,160 +7,125 @@ function scr_cheatcode(argument0)
+            obj_controller.disposition[4] = real(cheatcode_m_digits)
+            obj_controller.disposition[5] = real(cheatcode_m_digits)
+            obj_controller.disposition[6] = real(cheatcode_m_digits)
+            obj_controller.disposition[7] = real(cheatcode_m_digits)
+            obj_controller.disposition[8] = real(cheatcode_m_digits)
+            obj_controller.disposition[9] = real(cheatcode_m_digits)
+            obj_controller.disposition[10] = real(cheatcode_m_digits)
         }
-
-        else if string_count("dep", cheatcode_string) {
-            cheatcode_m_digits = clamp(cheatcode_m_digits, -100, 100)
-
-            if (cheatcode_string == "depimp")
-                obj_controller.disposition[2] = cheatcode_m_digits
-
-            else if (cheatcode_string == "depmec")
-                obj_controller.disposition[3] = cheatcode_m_digits
-
-            else if (cheatcode_string == "depinq")
-                obj_controller.disposition[4] = cheatcode_m_digits
-
-            else if (cheatcode_string == "depecc")
-                obj_controller.disposition[5] = cheatcode_m_digits
-
-            else if (cheatcode_string == "depeld")
-                obj_controller.disposition[6] = cheatcode_m_digits
-
-            else if (cheatcode_string == "depork")
-                obj_controller.disposition[7] = cheatcode_m_digits
-
-            else if (cheatcode_string == "deptau")
-                obj_controller.disposition[8] = cheatcode_m_digits
-
-            else if (cheatcode_string == "deptyr")
-                obj_controller.disposition[9] = cheatcode_m_digits
-
-            else if (cheatcode_string == "depcha")
-                obj_controller.disposition[10] = cheatcode_m_digits
-
-            else if (cheatcode_string == "depall") {
-                global.cheat_disp = 1
-                cheatyface = 1
-                obj_controller.disposition[2] = real(cheatcode_m_digits)
-                obj_controller.disposition[3] = real(cheatcode_m_digits)
-                obj_controller.disposition[4] = real(cheatcode_m_digits)
-                obj_controller.disposition[5] = real(cheatcode_m_digits)
-                obj_controller.disposition[6] = real(cheatcode_m_digits)
-                obj_controller.disposition[7] = real(cheatcode_m_digits)
-                obj_controller.disposition[8] = real(cheatcode_m_digits)
-                obj_controller.disposition[9] = real(cheatcode_m_digits)
-                obj_controller.disposition[10] = real(cheatcode_m_digits)
+    }
+    else if (cheatcode_string == "stc") {
+        cheatcode_digits = clamp(cheatcode_digits, 0, 100)
+        repeat cheatcode_digits
+        scr_add_stc_fragment()
+    } else if (string_count("event", cheatcode) == 1) {
+        with(obj_controller)
+        scr_random_event(1)
+    } else if (cheatcode_string == "recruit") {
+        var _start_pos = 0
+        var length = (array_length(obj_controller.recruit_name) - 1)

```

Comment:
the fix itself (changed `get_length` to `array_length`), all other changes are from the beautification pass